### PR TITLE
ROX-10075: Postgres directly scoped SAC for Get method

### PR DESCRIPTION
**Pull request by janisz**
## Description

SAC for Get method

## Testing Performed

CI

### CODE REVIEW

File: tools/generate-helpers/pg-table-bindings/store_test.go.tpl [hunk 1]
```diff
@@ -273,6 +273,39 @@ func (s *{{$namePrefix}}StoreSuite) TestSACExists() {
...
+func (s *{{$namePrefix}}StoreSuite) TestSACGet() {
+	objA := &{{.Type}}{}
+	s.NoError(testutils.FullInit(objA, testutils.UniqueInitializer(), testutils.JSONFieldsFilter))
+
+	objB := &{{.Type}}{}
+	s.NoError(testutils.FullInit(objB, testutils.UniqueInitializer(), testutils.JSONFieldsFilter))
+
+	withAllAccessCtx := sac.WithAllAccess(context.Background())
+	s.store.Upsert(withAllAccessCtx, objA)
+	s.store.Upsert(withAllAccessCtx, objB)
```
Comment (rhybrillou) [inline tools/generate-helpers/pg-table-bindings/store_test.go.tpl:286]: The test seems only to rely on objA

**Review by rhybrillou — APPROVED** [review]

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 2]
```diff
@@ -48,9 +48,9 @@ const (
 {{/* TODO(ROX-10624): Remove this condition after all PKs fields were search tagged (PR #1653) */}}
 {{- if gt (len $pks) 1 }}
         existsStmt = "SELECT EXISTS(SELECT 1 FROM {{.Table}} WHERE {{template "whereMatch" $pks}})"
+        getStmt = "SELECT serialized FROM {{.Table}} WHERE {{template "whereMatch" $pks}}"
```
Comment (md2119) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:51]: What is the approach for SAC when obj has more than primary key?

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 3]
```diff
@@ -528,8 +531,42 @@ func (s *storeImpl) Get(ctx context.Context, {{template "paramList" $pks}}) (*{{
...
+{{/* TODO(ROX-10624): Remove this condition after all PKs fields were search tagged (PR #1653) */}}
+{{- if eq (len $pks) 1 }}
+    q := search.ConjunctionQuery(
+    sacQueryFilter,
+    {{- range $idx, $pk := $pks}}
+        {{- if eq $pk.Name $singlePK.Name }}
+            search.NewQueryBuilder().AddDocIDs({{ $singlePK.ColumnName|lowerCamelCase }}).ProtoQuery(),
+        {{- else }}
+            search.NewQueryBuilder().AddExactMatches(search.FieldLabel("{{ $pk.Search.FieldName }}"), {{ $pk.ColumnName|lowerCamelCase }}).ProtoQuery(),
+        {{- end}}
```
Comment (md2119) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:561]: Why the fork? Can we have a unified approach use the search labels associated with the primary keys?

File: tools/generate-helpers/pg-table-bindings/store.go.tpl (hunk 2 above)
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/store.go.tpl:51]: Until `ROX-10624` is solved we use the old apprach for multiple keys after this we will use exact matches

File: tools/generate-helpers/pg-table-bindings/store.go.tpl (hunk 3 above)
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/store.go.tpl:561]: We can't. Most of single PK resources don't have search tag for ID. There are some exceptions listed below
```
# git grep  'sql:\"pk' -- '*.proto' | grep search

proto/storage/cluster.proto:    string                    id                               = 1 [(gogoproto.moretags) = 'search:"Cluster ID,hidden,store" sql:"pk"'];
proto/storage/cve.proto:    string                    id                    = 1 [(gogoproto.moretags) = 'search:"CVE ID,hidden" sql:"pk,id"'];  // This field is composite of cve and operating system.
proto/storage/deployment.proto:    string id                                  = 1 [(gogoproto.moretags) = 'search:"Deployment ID,store,hidden" sql:"pk"'];
proto/storage/image.proto:    string                          id                          = 4 [(gogoproto.moretags) = 'search:"Image Sha,store,hidden" sql:"pk"'];
proto/storage/image_component.proto:    string                         id               = 1 [(gogoproto.moretags) = 'search:"Component ID,store,hidden" sql:"pk,id"']; // This field is composite id over name, version, and operating system.
proto/storage/namespace_metadata.proto:    string                    id            = 1 [(gogoproto.moretags) = 'search:"Namespace ID" sql:"pk"'];
proto/storage/policy.proto:    string                      id                      = 1 [(gogoproto.moretags) = 'search:"Policy ID,store,hidden" sql:"pk"'];
proto/storage/process_indicator.proto:    string id                                      = 1 [(gogoproto.moretags) = 'search:"Process ID,store,hidden"  sql:"pk"'];
proto/storage/rbac.proto:    string id                               = 1 [(gogoproto.moretags) = 'search:"Role ID,hidden" sql:"pk"'];
proto/storage/rbac.proto:    string id                               = 1 [(gogoproto.moretags) = 'search:"Role Binding ID,hidden" sql:"pk"'];
proto/storage/secret.proto:    string              id               = 1 [(gogoproto.moretags) = 'search:"Secret ID,store,hidden" sql:"pk"'];
proto/storage/test.proto:    string key                 = 1 [(gogoproto.moretags) = 'sql:"pk,index=hash" search:"Test Key"'];
proto/storage/test.proto:  string key1                  = 1 [(gogoproto.moretags) = 'sql:"pk,id" search:"Test Key"'];
proto/storage/test.proto:  string id  = 1 [(gogoproto.moretags) = 'search:"Test Grandparent ID" sql:"pk"'];
proto/storage/test.proto:   string id        = 1 [(gogoproto.moretags) = 'search:"Test Parent1 ID" sql:"pk"'];
proto/storage/test.proto:  string id = 1 [(gogoproto.moretags) = 'search:"Test Child1 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test Grandchild1 ID" sql:"pk"'];
proto/storage/test.proto:  string id = 1 [(gogoproto.moretags) = 'search:"Test GGrandchild1 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test G2Grandchild1 ID" sql:"pk"'];
proto/storage/test.proto:  string id = 1 [(gogoproto.moretags) = 'search:"Test G3Grandchild1 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test Parent2 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test Child2 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test Parent3 ID" sql:"pk"'];
```

**Review by md2119 — APPROVED** [review]

**Comment by janisz** [comment]
@rhybrillou could you look again?

**Review by rhybrillou — APPROVED** [review]
Good job

**Pull request merged**